### ISSUE TITLE
Bug: The reference shoreline buffer is not being applied correctly in zoo

### ISSUE THREAD

**Issue by 2320sharon** [Doodleverse/CoastSeg#221]
In Zoo the reference shoreline buffer is not being applied correctly. Shorelines are being extracted even when they are outside the reference shoreline buffer as shown in the figure below. I have verified via the example below that the reference shoreline buffer is being plotted correctly, but it doesn't appear to be applied correctly.

What's weird though is that if I increase the size of the buffer more shorelines are being extracted as seen in the second figure.

![1984-11-01-15-03-39_L5](https://github.com/Doodleverse/CoastSeg/assets/61564689/3fd8576e-ac70-4d31-96fb-891e1c1fbfa1)
![1984-07-03-15-08-24_L5](https://github.com/Doodleverse/CoastSeg/assets/61564689/c533426f-788d-474e-814c-d72e0340f2ac)

**Comment by 2320sharon** [comment]
I figured out the cause of this bug at last!
When shorelines are extracted from the image the reference shoreline buffer is dilated by 5 pixels in both the `coastsat_package` workflow and the zoo workflow. The issue is the reference shoreline buffer shown in the plots was not dilated at all meaning that the reference shoreline buffer shown on the plots did not reflect the actual reference shoreline buffer used to extract the shoreline. 
```python
    # dilate the reference shoreline buffer by 5 pixels to mask out anything outside the reference shoreline buffer
    se = morphology.disk(5)
    im_ref_buffer_extra = morphology.binary_dilation(im_ref_buffer, se)
```
## Without Dilating Shoreline Buffer on Plot
- Reference Shoreline Buffer 24m
![2023-12-16-18-46-24_L9](https://github.com/Doodleverse/CoastSeg/assets/61564689/4f20e06d-f4e9-47de-85c9-80f3dd2b5063)

## With Dilating Shoreline Buffer on Plot
- Reference Shoreline Buffer 24m
- Dilated by 5 pixels to match what's done in the shoreline extraction functions
![2024-01-01-18-46-21_L9](https://github.com/Doodleverse/CoastSeg/assets/61564689/6f5910ea-06e9-41df-914c-2e5c61fad6ed)

**Comment by 2320sharon** [comment]
So this issue was bit more complicated than I previously thought. Here is what actually ended up being the solution.
I found out within `coastsat_package ` `find_wl_contours2` and `find_wl_contours1` that both of these functions were dilating the reference shoreline buffer by 5 pixels and keeping only the data within that region. The same was happening in the zoo workflows ` simplified_find_contours` function. This would normally be fine except the `SDS_shoreline.create_shoreline_buffer` had already dilated the shoreline reference buffer by 5 pixels meaning that the buffer was being dilated by a total of 10 pixels, but the plot only reflected the 5 pixel dilation. I modified coastsat_package and  ` simplified_find_contours` to no longer dilate the reference shoreline buffer after it was dilated.
